packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.native.test.ts

```diff
@@ -0,0 +1,67 @@
+import { getBaseBoxStyles, getResponsiveValue, getSpacingValue } from './getBaseBoxStyles';
+import type { BaseBoxProps } from './types';
+import paymentLightTheme from '~components/BladeProvider/__tests__/paymentLightTheme/paymentLightTheme';
+
+describe('getResponsiveValue', () => {
+  it('should return correctly for plain values', () => {
+    expect(getResponsiveValue('hello', 'base')).toBe('hello');
+    expect(getResponsiveValue('hello', 'xs')).toBe(undefined);
+  });
+
+  it('should correctly handle falsy values', () => {
+    expect(getResponsiveValue(undefined)).toBe(undefined);
+    // @ts-expect-error: intentional null to check the falsy values
+    expect(getResponsiveValue(null)).toBe(undefined);
+    expect(getResponsiveValue(0)).toBe(0);
+    expect(getResponsiveValue('')).toBe('');
+    expect(getResponsiveValue({ base: 3, s: 0 })).toBe(0);
+  });
+
+  it('should return responsive values in decreasing sequence starting from s token', () => {
+    expect(getResponsiveValue({ base: 'base-value', xs: 'xs-value', s: 'small-value' })).toBe(
+      'small-value',
+    );
+    expect(getResponsiveValue({ base: 'base-value', xs: 'xs-value' })).toBe('xs-value');
+    expect(getResponsiveValue({ base: 'base-value' })).toBe('base-value');
+    expect(getResponsiveValue({ m: 'medium-value' })).toBe(undefined);
```

Comment:
> now m is tablet and above?

Yup. For React Native, we start looking from `s` value. (Similar to `useBreakpointValue` hook's behaviour)

Reply:
+1, didn't understand why its returning undefined.

Reply:
I have added a comment for it in that function

https://github.com/razorpay/blade/blob/6e23c240e6bab4133e6b602b673ef9465e2bd25b/packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.ts#L24-L46

TLDR is `m` is a desktop style so it should be undefined in react native because we haven't added any styles for mobile.


This is kind of similar to this media query

```css
.something {
  /* No base styles defined */
}

@media desktop {
   .something {
      display: 'none'
   }
}
```

Here the `display` property is not defined for base styles or mobile styles so it should translate to `undefined` (non-existing value) in react native too.